harness/determined/launch/autodeepspeed.py

```diff
@@ -0,0 +1,239 @@
+"""
+autodeepspeed.py is the launch layer for DeepSpeedTrial in Determined.
+
+It launches the entrypoint script using DeepSpeed's launch process.
+"""
+import argparse
+import logging
+import os
+import pathlib
+import socket
+import subprocess
+import sys
+import time
+from typing import Dict, List, cast
+
+from deepspeed.launcher.runner import DEEPSPEED_ENVIRONMENT_NAME, DLTS_HOSTFILE
```

Comment:
This smells like an internal API from deepspeed to me.  Is there an API to explicitly set what file deepspeed reads for the environment?

Reply:
yes, the filepath is now passed in as an argument to our call of the deepspeed launcher.